packages/core/admin/admin/src/components/Roles/ContentTypeCollapse/Collapse/index.js

```diff
@@ -66,69 +124,84 @@ const Collapse = ({
+        onClick={onClickToggle}
+        someChecked={hasSomeActionsSelected}
+        value={hasAllActionsSelected}
+      >
+        <Chevron paddingLeft={2}>{isActive ? <Up /> : <Down />}</Chevron>
+      </RowLabelWithCheckbox>
+
+      <Row style={{ flex: 1 }}>
+        {checkboxesActions.map(
+          ({
+            actionId,
+            hasConditions,
+            hasAllActionsSelected,
+            hasSomeActionsSelected,
+            isDisplayed,
+            isParentCheckbox,
+            checkboxName,
+          }) => {
+            if (!isDisplayed) {
+              return <HiddenAction key={actionId} />;
+            }
+
+            if (isParentCheckbox) {
+              return (
+                <Cell key={actionId} justifyContent="center" alignItems="center">
+                  {hasConditions && <TinyDot />}
+                  <Checkbox
                     disabled={isFormDisabled || IS_DISABLED}
-                    hasConditions={hasConditions}
                     name={checkboxName}
-                    onChange={onChangeParentCheckbox}
-                    someChecked={hasSomeActionsSelected}
+                    // Keep same signature as packages/core/admin/admin/src/components/Roles/Permissions/index.js l.91
+                    onValueChange={value =>
+                      onChangeParentCheckbox({
+                        target: {
+                          name: checkboxName,
+                          value,
+                        },
+                      })}
+                    indeterminate={hasSomeActionsSelected}
                     value={hasAllActionsSelected}
                   />
-                );
-              }
+                </Cell>
+              );
+            }
 
-              return (
-                <CheckboxWithCondition
-                  key={actionId}
+            return (
+              <Cell key={actionId} justifyContent="center" alignItems="center">
+                {hasConditions && <TinyDot />}
+                <Checkbox
                   disabled={isFormDisabled || IS_DISABLED}
-                  hasConditions={hasConditions}
+                  indeterminate={hasConditions}
                   name={checkboxName}
-                  onChange={onChangeSimpleCheckbox}
+                  // Keep same signature as packages/core/admin/admin/src/components/Roles/Permissions/index.js l.91
+                  onValueChange={value =>
+                    onChangeSimpleCheckbox({
+                      target: {
+                        name: checkboxName,
+                        value,
+                      },
+                    })}
                   value={hasAllActionsSelected}
                 />
-              );
-            }
-          )}
-        </Flex>
+              </Cell>
+            );
+          }
+        )}
+      </Row>
+      <Box paddingRight={6}>
         <ConditionsButton
-          isRight
           onClick={handleToggleModalIsOpen}
```

Comment:
When the modal is closed, I have to click two times on this button to re open the modal.